src/Futurelink.vue

```diff
@@ -43,11 +43,25 @@
 
         loaded.push(href);
 
-        this.$emit('preload', href);
-
         const resolved = this.$router.resolve(href);
         const matched = resolved.resolved.matched;
-        this.preloadComponent = matched[matched.length - 1].components.default;
+        const route = matched[matched.length - 1];
+        // Only preload if a route was found.
+        if (route) {
```

Comment:
i'd prefer it if this were negated and returned early instead - https://github.com/felixge/node-style-guide#return-early-from-functions